src/Sylius/Component/Core/Promotion/Action/FixedDiscountPromotionActionCommand.php

```diff
@@ -86,14 +101,30 @@ public function execute(PromotionSubjectInterface $subject, array $configuration
         }
 
         $splitPromotion = $this->proportionalDistributor->distribute($itemsTotals, $promotionAmount);
-        $this->unitsPromotionAdjustmentsApplicator->apply($subject, $promotion, $splitPromotion);
+        $this->adjustmentsApplicator->apply($subject, $promotion, $splitPromotion);
 
         return true;
     }
 
     /**
      * {@inheritdoc}
      */
+    public function revert(PromotionSubjectInterface $subject, array $configuration, PromotionInterface $promotion)
+    {
+        if (!$subject instanceof OrderInterface) {
+            throw new UnexpectedTypeException($subject, OrderInterface::class);
```

Comment:
What would you say for using `Assert::instanceOf(...)` instead? Here and in similar places below.